src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.tsx

```diff
@@ -26,24 +26,36 @@ import {t} from 'Util/LocalizerUtil';
 import {styles} from './CertificateDetailsModal.styles';
 
 const COPY_MESSAGE_TIMEOUT = 3000;
+const CERTIFICATE_NAME = 'certificate.pem';
 
 export interface CertificateDetailsModalProps {
   certificate: string;
   onClose: () => void;
-  downloadCertificate: () => void;
 }
 
-export const CertificateDetailsModal = ({certificate, onClose, downloadCertificate}: CertificateDetailsModalProps) => {
+export const CertificateDetailsModal = ({certificate, onClose}: CertificateDetailsModalProps) => {
   const [isTextCopied, setIsTextCopied] = useState(false);
   const [isDownloading, setIsDownloading] = useState(false);
 
   const onDownload = () => {
     setIsDownloading(true);
-    downloadCertificate();
 
-    setTimeout(() => {
+    const downloadLink = document.createElement('a');
```

Comment:
suggestion: There's `downloadFile` function in `util.ts` file, maybe we could use it (or extend it to match this use case if it's missing something).